calibrate.py

```diff
@@ -114,6 +199,186 @@ def parse_args():
+        #     self.earliest_ts = {name: msg.getTimestampDevice()}
+
+        # print('Queues: ', end='')
+        # for name in self.queues.keys():
+        #     print('\t: ', name, end='')
+        #     print(self.queues[name], end=', ')
+        #     print()
+        # print()
+
+    def get_synced(self, displayMinimum):
+
+        # Atleast 3 messages should be buffered
+        min_len = min([len(queue) for queue in self.queues.values()])
+        if min_len == 0:
+            print('Status:', 'exited due to min len == 0', self.queues)
+            return None
+
+        # initializing list of list 
+        queue_lengths = []
+        for name in self.queues.keys():
+            queue_lengths.append(range(0, len(self.queues[name])))
+        permutations = list(itertools.product(*queue_lengths))
+        # print ("All possible permutations are : " +  str(permutations))
+
+        # Return a best combination after being atleast 3 messages deep for all queues
+        min_ts_diff = None
+        for indicies in permutations:
+            tmp = {}
+            i = 0
+            for n in self.queues.keys():
+                tmp[n] = indicies[i]
+                i = i + 1
+            indicies = tmp
+
+            acc_diff = 0.0
+            min_ts = None
+            for name in indicies.keys():
+                msg = self.queues[name][indicies[name]]
+                if min_ts is None:
+                    min_ts = msg.getTimestampDevice().total_seconds()
+            for name in indicies.keys():
+                msg = self.queues[name][indicies[name]]
+                acc_diff = acc_diff + abs(min_ts - msg.getTimestampDevice().total_seconds())
+
+            # Mark minimum
+            if min_ts_diff is None or (acc_diff < min_ts_diff['ts'] and abs(acc_diff - min_ts_diff['ts']) > 0.0001):
+                min_ts_diff = {'ts': acc_diff, 'indicies': indicies.copy()}
+                if displayMinimum:
+                    print('new minimum:', min_ts_diff, 'min required:', self.min_diff_timestamp)
+
+            if min_ts_diff['ts'] < self.min_diff_timestamp:
+                # Check if atleast 5 messages deep
+                min_queue_depth = None
+                for name in indicies.keys():
+                    if min_queue_depth is None or indicies[name] < min_queue_depth:
+                        min_queue_depth = indicies[name]
+                if min_queue_depth >= self.queue_depth:
+                    # Retrieve and pop the others
+                    synced = {}
+                    for name in indicies.keys():
+                        synced[name] = self.queues[name][min_ts_diff['indicies'][name]]
+                        # pop out the older messages
+                        for i in range(0, min_ts_diff['indicies'][name]+1):
+                            self.queues[name].popleft()
+
+                    print('Returning synced messages with error:', min_ts_diff['ts'], min_ts_diff['indicies'])
+                    return synced
```

Comment:
make this part of `TraceLevel` or make it part of `displayMinimum` by replacing the `displayMinimum` name to `enableDebugMessageSync`

Reply:
I changed displayMinimum into enableDebugMessageSync in all cases.